tools/clang/tools/dxcompiler/dxcompilerobj.cpp

```diff
@@ -2081,6 +2092,38 @@ class DxcCompiler : public IDxcCompiler, public IDxcLangExtensions, public IDxcC
         action.EndSourceFile();
         outStream.flush();
       }
+      else if (rootSigMajor) {
+        HLSLRootSignatureAction action(
+            compiler.getCodeGenOpts().HLSLEntryFunction, rootSigMajor,
+            rootSigMinor);
+        FrontendInputFile file(utf8SourceName.m_psz, IK_HLSL);
+        action.BeginSourceFile(compiler, file);
+        action.Execute();
+        action.EndSourceFile();
+        outStream.flush();
+        // Don't do work to put in a container if an error has occurred
+        bool compileOK = !compiler.getDiagnostics().hasErrorOccurred();
+        if (compileOK) {
+          auto rootSigHandle = action.takeRootSigHandle();
+
+          CComPtr<AbstractMemoryStream> pContainerStream;
+          IFT(CreateMemoryStream(pMalloc, &pContainerStream));
+          SerializeDxilContainerForRootSignature(rootSigHandle.get(),
+                                                 pContainerStream);
+
+          pOutputBlob.Release();
+          IFT(pContainerStream.QueryInterface(&pOutputBlob));
+
+          CComPtr<IDxcBlob> pTargetBlob;
+          if (m_pDxcContainerEventsHandler != nullptr) {
+            HRESULT hr = m_pDxcContainerEventsHandler->OnDxilContainerBuilt(
```

Comment:
Probably do not need to call the event here. I think it was intended as a callback after compilation succeeded.